### PR TITLE
Move sets.Set to common location

### PR DESCRIPTION
**Pull request by nmittler**
Also cleaning it up a bit and modifying a number of callsites to use it.

**Please provide a description of this PR:**

### CODE REVIEW

**Review by douglas-reid — APPROVED** [review]
Approving P&T changes.

**Review by howardjohn — APPROVED** [review]

**Review by therealmitchconnors — APPROVED** [review]

File: istioctl/pkg/writer/envoy/configdump/route.go [hunk 1]
```diff
@@ -111,8 +111,8 @@ func describeRouteDomains(domains []string) string {
 }
 
 func unexpandDomains(domains []string) []string {
-	unique := sets.NewSet(domains...)
-	shouldDelete := sets.NewSet()
+	unique := sets.NewWith(domains...)
```
Comment (hzxuzhonghu) [inline istioctl/pkg/writer/envoy/configdump/route.go:114]: Why add a additional function

File: pkg/util/sets/string.go [hunk 2]
```diff
@@ -18,23 +18,38 @@ import "sort"
...
+}
+
+// Insert adds the item to the set.
+func (s Set) Insert(item string) Set {
+	s[item] = struct{}{}
+	return s
+}
+
+// InsertAll adds items to the set.
+func (s Set) InsertAll(items ...string) Set {
```
Comment (hzxuzhonghu) [inline pkg/util/sets/string.go:38]: We donot need to change it

File: pkg/util/sets/string.go [hunk 3]
```diff
@@ -18,23 +18,38 @@ import "sort"
...
-// Delete removes items from the set.
-func (s Set) Delete(items ...string) Set {
+// Delete removes the item from the set.
+func (s Set) Delete(item string) Set {
+	delete(s, item)
+	return s
+}
+
+// DeleteAll removes items from the set.
+func (s Set) DeleteAll(items ...string) Set {
```
Comment (hzxuzhonghu) [inline pkg/util/sets/string.go:52]: ditto

File: pkg/util/sets/string.go [hunk 4]
```diff
@@ -60,7 +75,7 @@ func (s Set) Merge(s2 Set) Set {
 // s2 = {a1, a2, a4, a5}
 // s.Union(s2) = s2.Union(s) = {a1, a2, a3, a4, a5}
 func (s Set) Union(s2 Set) Set {
-	result := NewSet()
+	result := NewWith()
```
Comment (ramaraochavali) [inline pkg/util/sets/string.go:78]: This can be just New not NewWith

File: pkg/util/sets/string.go [hunk 5]
```diff
@@ -77,7 +92,7 @@ func (s Set) Union(s2 Set) Set {
 // s.Difference(s2) = {a3}
 // s2.Difference(s) = {a4, a5}
 func (s Set) Difference(s2 Set) Set {
-	result := NewSet()
+	result := NewWith()
```
Comment (ramaraochavali) [inline pkg/util/sets/string.go:95]: same here and other usages

File: pkg/util/sets/string.go (hunk 4 above)
Comment (hzxuzhonghu) [reply pkg/util/sets/string.go:78]: yep

File: istioctl/pkg/writer/envoy/configdump/route.go (hunk 1 above)
Comment (nmittler) [reply istioctl/pkg/writer/envoy/configdump/route.go:114]: Avoiding a needless array allocation.

File: pkg/util/sets/string.go (hunk 2 above)
Comment (nmittler) [reply pkg/util/sets/string.go:38]: Also avoiding needless array allocation

File: pkg/util/sets/string.go (hunk 3 above)
Comment (nmittler) [reply pkg/util/sets/string.go:52]: same thing here.

**Pull request merged**